include/boost/geometry/strategies/distance_result.hpp

```diff
@@ -78,12 +78,7 @@ struct distance_result_strategy_type<Geometry1, Geometry2, Strategy, true>
 {};
 
 
-template
```

Comment:
It's because this was a part of the workaround for VS2015. With this compiler Boost.Variant disables variadic templates and emulates them instead. This means that at the end of the list of arguments there are types `boost::detail::variant::null` or something like that which are not geometries. `AreGeometries` was used to ignore them with VS2015 by setting the distance result to `int` for them. Now the adaptation for Boost.Variant handles it so this workaround including `AreGeometries` is not needed anymore.